browser.js

```diff
@@ -48,13 +48,13 @@ function pbkdf2Sync (password, salt, iterations, keylen, digest) {
   if (!Buffer.isBuffer(password)) password = new Buffer(password, 'binary')
   if (!Buffer.isBuffer(salt)) salt = new Buffer(salt, 'binary')
 
+  var destPos = 0
   var hLen
   var l = 1
   var DK = new Buffer(keylen)
   var block1 = new Buffer(salt.length + 4)
   salt.copy(block1, 0, 0, salt.length)
 
-  var r
   var T
 
   for (var i = 1; i <= l; i++) {
```

Comment:
@fanatid unfortunately I don't know a way to get the hash output size before entering the loop... the API's might have changed though.